weblogic-azure-aks/src/main/arm/scripts/setupDBConnections.sh

```diff
@@ -0,0 +1,322 @@
+        "${wlsDomainUID}" \
+        "${dbSecretName}" \
+        "${optTypeUpdate}"
+}
+
+function apply_datasource_to_domain() {
+    echo "apply datasoure"
+    # get domain configurations
+    domainConfigurationJsonFile=$scriptDir/domain.json
+    kubectl -n ${wlsDomainNS} get domain ${wlsDomainUID} -o json >${domainConfigurationJsonFile}
+
+    restartVersion=$(cat ${domainConfigurationJsonFile} | jq '. | .spec.restartVersion' | tr -d "\"")
+    secretList=$(cat ${domainConfigurationJsonFile} | jq -r '. | .spec.configuration.secrets')
+    restartVersion=$((restartVersion+1))
+
+    echo "current secrets: ${secretList}"
+    if [[ "${secretList}" != "null" ]];then
+        secretList=$(cat ${domainConfigurationJsonFile} | jq -r '. | .spec.configuration.secrets[]')
+        secretStrings="["
+        index=0;
+        for item in $secretList; do
+            if [[ "${item}" == "${dbSecretName}" ]]; then
+                continue
+            fi
+
+            if [ $index -eq 0 ];then
+                secretStrings="${secretStrings}\"${item}\","
+            else
+                secretStrings="${secretStrings}\"${item}\","
+            fi
+
+            index=$((index+1))
+        done
+
+        secretStrings="${secretStrings}\"${dbSecretName}\"]"
+    else
+        secretStrings="[\"${dbSecretName}\"]"
+    fi
+
+    echo "secrets: ${secretStrings}"
+
+    # apply the configmap
+    # apply the secret
+    # restart the domain
+    timestampBeforePatchingDomain=$(date +%s)
+    kubectl -n ${wlsDomainNS} patch domain ${wlsDomainUID} \
+        --type=json \
+        -p '[{"op": "replace", "path": "/spec/restartVersion", "value": "'${restartVersion}'" }, {"op": "replace", "path": "/spec/configuration/model/configMap", "value":'${wlsConfigmapName}'}, {"op": "replace", "path": "/spec/configuration/secrets", "value": '${secretStrings}'}]'
+}
+
+function remove_datasource_from_domain() {
+    echo "rollback datasoure"
+    # get domain configurations
+    domainConfigurationJsonFile=$scriptDir/domain.json
+    kubectl -n ${wlsDomainNS} get domain ${wlsDomainUID} -o json >${domainConfigurationJsonFile}
+
+    restartVersion=$(cat ${domainConfigurationJsonFile} | jq '. | .spec.restartVersion' | tr -d "\"")
+    secretList=$(cat ${domainConfigurationJsonFile} | jq -r '. | .spec.configuration.secrets')
+    restartVersion=$((restartVersion+1))
+
+    echo "current secrets: ${secretList}"
+    if [[ "${secretList}" != "null" ]];then
+        secretList=$(cat ${domainConfigurationJsonFile} | jq -r '. | .spec.configuration.secrets[]')
+        secretStrings="["
+        index=0;
+        for item in $secretList; do
+            ret=$(kubectl -n ${wlsDomainNS} get secret | grep "${item}")
+            # the secret should have been deleted.
+            if [ -z "${ret}" ]; then
+                continue
+            fi
+
+            if [ $index -eq 0 ];then
+                secretStrings="${secretStrings}\"${item}\","
+            else
```

Comment:
How are the branches different? Maybe it's just my eyes, but the if and else branches look identical.

Reply:
This is a bug. The branches is to generate the secret array string with/without comma for different cases. I found `kubectl patch` works fine with values like `["secret1","secret2",]`, unfortunately, I didn't find this bug during e2e testing.

Fixed it in https://github.com/galiacheng/weblogic-azure/commit/17537a088bc845f9cfe9d21f8d55949a5bcab4ea